tools/myst/enc/clock.c

```diff
@@ -10,23 +10,39 @@
 
 static long _realtime0 = 0;
 static long _monotime0 = 0;
+// The address of this pointer should be 8-byte aligned
+// since it points to host address
 static volatile long* _monotime_now = 0;
 static long _realtime_delta = 0;
 static long enc_clock_res = 0;
 
+void oe_memcpy_aligned(void* dest, const void* src, size_t count);
+
 int myst_setup_clock(struct clock_ctrl* ctrl)
 {
+    // ctrl is a host address.
+    // Store a copy of its content in enclave address to
+    // avoid reading host memory multiple times.
+    struct clock_ctrl* ctrl_enclave = NULL;
     int ret = -1;
     if (ctrl != NULL)
     {
+        if (sizeof(struct clock_ctrl) % 8 != 0 || (uint64_t)ctrl % 8 != 0)
```

Comment:
The runtime check against the static size is not necessary. Use this approach
https://github.com/deislabs/mystikos/blob/ddccafad171c9c24accb2a4b7605e937c319909a/include/myst/luks.h#L44
in the clock.h to check the size.